### PR TITLE
[Hub Generated] Review request for Microsoft.Resources to add version stable/2023-07-01

### PR DESCRIPTION
**Pull request by gary-x-li**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/gary-x-li-resources-Microsoft.Resources-2023-07-01...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Comment by rkmanda** [comment]
Please provide explanations for the Lintdiff, Avocado and Model validation failures

**Comment by rkmanda** [comment]
There are other required checks failing as well. Please take a look

**Comment by gary-x-li** [comment]
> Please provide explanations for the Lintdiff, Avocado and Model validation failures

@rkmanda 
- Lintdiff: To fix all the violations, it would require some work including backend code changes. We'll create a task for this work.
- Avocado: To fix the only violation, I think we'll need to make some extensive changes to the json file to follow the [Folder Structure for Service Group](https://github.com/Azure/azure-rest-api-specs/blob/main/README.md#folder-structure-for-service-group). We'll create a task for this work.
- Model validation: Most instances are [XMS_EXAMPLE_NOTFOUND_ERROR](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Semantic-and-Model-Violations-Reference.md#XMS_EXAMPLE_NOTFOUND_ERROR). Currently we have the [examples](https://github.com/Azure/azure-rest-api-specs/tree/4d9fece121bbac80da74f786b9a94be5960eaf3c/specification/resources/resource-manager/Microsoft.Resources/stable/2023-07-01/examples) folder which contains various examples. We'll create a task for this work.

To sum up, this PR simply copied the previous api version to re-release a small feature. All the check failures are not introduced by this PR. We'll create work items and fix them ASAP. Can you please merge it for now?

**Comment by rkmanda** [comment]
Please apply suppressions for all failures with a link to the work items in each

https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process

**Comment by rkmanda** [comment]
once you are done, place this back in the merge queue by adding the mergeRequested label

**Review by rkmanda — APPROVED** [review]

**Comment by kazrael2119** [comment]
please fix ```SemanticValidation``` error to let sdk generation work

**Comment by ms-henglu** [comment]
Please fix the Swagger Avocado error and get approval for `SdkBreakingChange-JavaScript`.

And about the suppression, since it's mentioned they're existing errors, would you please also add path to limit the suppressed scope.

**Comment by kazrael2119** [comment]
this js sdk breaking is not caused by this pr

**Comment by qiaozha** [comment]
Remove the ARMSignedOff label as the ARMChangesRequested label should not co-exist with signed off label and changes requested is latest added.

**Comment by konrad-jamrozik** [comment]
Approving Avocado per [this discussion](https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1692730166749?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1692229279894&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1692730166749).

**Review by rkmanda — APPROVED** [review]
